File: Backend/src/main/java/com/smartamenities/backend/dto/AmenityLiveStatusResponse.java
package com.smartamenities.backend.dto;

import com.smartamenities.backend.model.AmenityStatus;
import com.smartamenities.backend.model.AmenityStatusReason;
import com.smartamenities.backend.model.OccupancyStatus;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * DTO for GET /amenities/live-status response.
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class AmenityLiveStatusResponse {
    private String amenityId;
    private AmenityStatus status;
    private AmenityStatusReason statusReason;
    private Integer waitTimeMinutes;
    private Integer stallsAvailable;
    private OccupancyStatus occupancyStatus;
}

File: Mobile/app/src/main/java/com/smartamenities/app/model/AmenityLiveStatusResponse.kt
package com.smartamenities.app.model

data class AmenityLiveStatusResponse(
    val amenityId: String,
    val status: String? = null,
    val statusReason: String? = null,
    val waitTimeMinutes: Int? = null,
    val stallsAvailable: Int? = null,
    val occupancyStatus: String? = null,
    val selectable: Boolean? = null
)

